eth/vm/base.py

```diff
@@ -440,7 +495,15 @@ def pack_block(self, block: BlockAPI, *args: Any, **kwargs: Any) -> BlockAPI:
             )
 
         header: BlockHeaderAPI = block.header.copy(**kwargs)
-        packed_block = block.copy(uncles=uncles, header=header)
+
+        if hasattr(block, "withdrawals") and block.withdrawals not in (None, ()):
```

Comment:
**Note for refactoring:** This was made to work, re-think this logic a bit. Not just here, but this sort of logic appears in a few other places.